ec/resources/waf/access_rule.go

```diff
@@ -457,18 +447,214 @@ func ResourceAccessRuleCreate(ctx context.Context, d *schema.ResourceData, m int
 	log.Printf("[INFO] Successfully created WAF Access Rule: %+v", resp)
 
 	d.SetId(resp.ID)
+	ResourceAccessRuleRead(ctx, d, m)
 
 	return diags
 }
 
 func ResourceAccessRuleRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
 	var diags diag.Diagnostics
+	accountNumber := d.Get("account_number").(string)
+	ruleID := d.Id()
+	config := m.(**api.ClientConfig)
+	(*config).AccountNumber = accountNumber
+	wafService, err := buildWAFService(**config)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	log.Printf("[INFO] Reading WAF Access Rule Id %s for Account >> %s", accountNumber, ruleID)
+	resp, err := wafService.GetAccessRuleByID(accountNumber, ruleID)
+	if err != nil {
+		d.SetId("")
+		return diag.FromErr(err)
+	}
+	log.Printf("[INFO] Retrieved Rule: %+v", resp)
```

Comment:
As mentioned above, Go objects with pointer properties need to be printed manually, otherwise the log output will have memory addresses for those pointers

Reply:
Same as above, we should address Read method in Dashan's code.